test/unit/query/builder.js

```diff
@@ -3371,4 +3371,64 @@ describe("QueryBuilder", function() {
     })
   });
 
+
+  it('Any undefined binding in a SELECT query should throw an error', function() {
+    var expectedErrorMessageContains = 'Undefined binding(s) detected when compiling SELECT query:'; //This test is not for asserting correct queries
+    var qbuilders = [
+      qb().from('accounts').where({Login: void 0}).select(),
+      qb().from('accounts').where('Login', void 0).select(),
+      qb().from('accounts').where('Login', '>=', void 0).select(),
+      qb().from('accounts').whereIn('Login', ['test', 'val', void 0]).select(),
+      qb().from('accounts').where({Login: ['1', '2', '3', void 0]}),
+      qb().from('accounts').where({Login: {Test: '123', Value: void 0}}),
+      qb().from('accounts').where({Login: ['1', ['2', [void 0]]]})
+    ];
+    qbuilders.forEach(function(qbuilder) {
+      try {
```

Comment:
Probably you could use http://chaijs.com/api/bdd/#method_throw here

```js
expect(function () {
... throwing code ...
}).to.throw(expected error message or regex checking error or something like that);
```

Do you think we should test here also for example if `.select(function () { this.where('id', undefined); })` or `.select(['col', undefined])` etc. works?